core/store/ChainStore/ChainStore.go

```diff
@@ -821,13 +821,14 @@ func (bd *ChainStore) persist(b *Block) error {
 			}
 
 			// find Transactions[i].UTXOInputs[index].ReferTxOutputIndex and delete it
-			for k := 0; k < len(unspents[txhash]); k++ {
-				if unspents[txhash][k] == uint16(b.Transactions[i].UTXOInputs[index].ReferTxOutputIndex) {
-					unspents[txhash] = append(unspents[txhash], unspents[txhash][:k]...)
-					unspents[txhash] = append(unspents[txhash], unspents[txhash][k+1:]...)
-					break
+			txunspent := make([]uint16, 0, len(unspents[txhash]) - 1)
+			for _, outputIndex := range unspents[txhash]{
+				if outputIndex == uint16(b.Transactions[i].UTXOInputs[index].ReferTxOutputIndex){
+					continue
 				}
+				txunspent = append(txunspent, outputIndex)
```

Comment:
since the entries in unspents do not needed to be sorted ,  just a swap operation is enough for efficiency.
```rust
len := len(unspents[txhash])
unspents[txhash][k] = unspents[txhash][len-1]
unspents[txhash] =  unspents[txhash][:len-1]
```